### ISSUE TITLE
No png support on NixOS

### ISSUE THREAD

**Issue by xzfc** [DeadSix27/waifu2x-converter-cpp#123]
On NixOS, opencv has stripped build information, and waifu2x-converter-cpp thinks some output formats (including PNG) aren't supported, despite the fact that opencv still can handle them.

Not sure if this should be fixed on the waifu2x-converter-cpp side or on the NixOS side.

```
./waifu2x-converter-cpp --list-opencv-formats
This is a list of supported formats, it depends on which formats opencv has been built with.
	 BMP -> Yes
	 DIB -> Yes
	 EXR -> Yes
	 HDR -> Yes
	 JP2 -> No
	 JPE -> No
	JPEG -> No
	 JPG -> No
	 PBM -> Yes
	 PGM -> Yes
	 PIC -> Yes
	 PNG -> No
	 PNM -> Yes
	 PPM -> Yes
	 PXM -> Yes
	 RAS -> Yes
	  SR -> Yes
	 TIF -> No
	TIFF -> No
	WEBP -> No
```

```
python2.7 -c 'import cv2; print cv2.getBuildInformation()'
(build info elided)
```

https://github.com/DeadSix27/waifu2x-converter-cpp/blob/b4566c0fb33e716cb092cbf6a557137b53acbfac/src/main.cpp#L402-L422

**Comment by DeadSix27** [comment]
The detection of formats is somewhat hacky, as opencv doesn't seem to provide a proper way to query that information apart from the build configuration info.. so It's more on our side than NixOS's I'd say.
But why would they strip it anyway... 

I can't think of a good way to go about this though.. 

Easiest thing would be to add a switch to force support of all formats for edge cases like this..

**Comment by xzfc** [comment]
> But why would they strip it anyway...

It's a trick to reduce the number of runtime dependencies.
https://github.com/NixOS/nixpkgs/commit/c8a2533516b4174455601572c80298d0c23b08c8

**Comment by DeadSix27** [comment]
Another way could be to define the stuff upon compilation... so package maintainers on NixOS could have it build with said formats forced-on .. I'll add that I guess.

**Comment by xzfc** [comment]
It seems you can use `cvHaveImageWriter` to detect formats.

```c
#include <opencv2/opencv.hpp>
#include <iostream>
int main() {
	std::cout << cvHaveImageWriter(".png") << "\n";
	std::cout << cvHaveImageWriter(".mp4") << "\n";
}
```

```
$ ./a.out                              
1
0
```

**Comment by DeadSix27** [comment]
Nice.

**Comment by DeadSix27** [comment]
Fixed via #71 
Note that CV4+ is required now.
